react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js

```diff
@@ -1,64 +1,37 @@
-import React, { Component, Fragment } from 'react';
+import React, { useEffect, Fragment } from 'react';
 import Spinner from 'react-spinkit';
+import { useSelector, useDispatch } from 'react-redux';
 
-import api from '~services/MatchesService';
+import actionsCreators from '~redux/game/actions';
 
 import styles from './styles.module.scss';
 
-class Matches extends Component {
-  state = {
-    matches: [],
-    loading: true
-  };
-
-  componentDidMount() {
-    setTimeout(() => {
-      this.getMatches();
-    }, 3000);
-  }
-
-  // recomendacioón de alejo
-  parseResponse = response => {
-    const { ok, data, problem, status } = response;
-    if (ok) {
-      return data;
-    }
-    throw new Error({ ...problem, ...status });
-  };
-
-  getMatches = async () => {
-    try {
-      const response = await api.getMatches();
-      const data = this.parseResponse(response);
-      this.setState({
-        matches: data,
-        loading: false
-      });
-    } catch (error) {
-      console.error('error: ', error);
-    }
-  };
-
-  render() {
-    const { matches, loading } = this.state;
-    return (
-      <Fragment>
-        {loading ? (
-          <Spinner name="double-bounce" />
-        ) : (
-          <ol>
-            {matches.map(match => (
+function Matches() {
+  const dispatch = useDispatch();
+  const { loading, matches } = useSelector(state => state.game);
+
+  useEffect(() => {
+    dispatch(actionsCreators.getMatches());
```

Comment:
Again, beware of the times this code is executed. I think you might need to pass a second argument so that `useEffect` would know [whether it has to run or not](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects).